mantidproject/mantid
# Harmonise plotting of workspaces from right-button click

Until this, surface and contour plotting was done in quite a different way to the plotting of spectra with or without errors in simple 1D, waterfall or tiled plot. Now they should all be accessible from a `Plot Spectrum Advanced` right-click menu option and basic plot spectrum functionality should be available as before.

The issue #18159 shows this in more detail.

**To test:**

The issue shows a python script and file to load that would be useful in testing also use your own file or scripts as you see fit.

Test the right click menu for `Plot Spectrum`, `Plot Spectrum with Errors` and `Plot Spectrum Advanced` in the various options available. It has been noted that the legend may not always appear for a waterfall plot #19208 and when 1D or waterfall plots are done with log values instead of workspace index, the items may not be shown in numerical order in the legend.

Fixes #18159.


TO DO

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
